sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SchemaHelper.scala

```diff
@@ -28,6 +33,61 @@ import org.apache.spark.util.Utils
 /**
  * Helper classes for reading/writing state schema.
  */
+sealed trait ColumnFamilySchema extends Serializable {
+  def jsonValue: JValue
+
+  def json: String
+
+  def columnFamilyName: String
+}
+
+case class ColumnFamilySchemaV1(
+    columnFamilyName: String,
+    keySchema: StructType,
+    valueSchema: StructType,
+    keyStateEncoderSpec: KeyStateEncoderSpec,
+    multipleValuesPerKey: Boolean,
+    valueEncoder: StructType,
+    userKeyEncoder: Option[StructType] = None) extends ColumnFamilySchema {
+  def jsonValue: JValue = {
+    ("columnFamilyName" -> JString(columnFamilyName)) ~
+      ("keySchema" -> JString(keySchema.json)) ~
+      ("valueSchema" -> JString(valueSchema.json)) ~
+      ("keyStateEncoderSpec" -> keyStateEncoderSpec.jsonValue) ~
+      ("multipleValuesPerKey" -> JBool(multipleValuesPerKey)) ~
```

Comment:
Do we need these 2 here ?

Reply:
I think we do need these, right? These shouldn't change between restarts

Reply:
Discussed offline

Reply:
We need to remove these 2 ?

Reply:
Doesn't multipleValuesPerKey tell us whether it's a ListState or not? This is the only thing distinguishing from ListState and ValueState

Reply:
lets reintroduce valueSchema as discussed